provider/provider.go

```diff
@@ -0,0 +1,86 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"net"
+	"net/rpc"
+	"os"
+	"os/exec"
+	"path"
+
+	"github.com/lukedmor/conduit_tiny/common"
+)
+
+type Executor struct{}
+
+func (xr *Executor) Execute(x *common.Executable, r *common.ExecutionReply) error {
```

Comment:
In addition to RPC, consider offering ssh/scp (e.g. once we implement docker containers, can send an ssh key to client upon instantiation)